src/main/java/com/avispl/symphony/dal/avdevices/wireless/presentation/kramer/viaconnectpro/VIAConnectProCommunicator.java

```diff
@@ -0,0 +1,1432 @@
+
+	/**
+	 * Check if {@link VIAConnectProCommunicator#controlProperty(ControllableProperty)} have just finished
+	 * And {@link VIAConnectProCommunicator#getMultipleStatistics()} is about to be executed.
+	 */
+	private boolean isCachedControlling;
+
+	/**
+	 * Check if user is controlling StreamingControl
+	 */
+	private boolean isStreamingControl = false;
+
+	/**
+	 * Store previous username
+	 */
+	private String previousUserName;
+
+	/**
+	 * ReentrantLock to prevent telnet session is closed when adapter is retrieving statistics from the device.
+	 */
+	private final ReentrantLock reentrantLock = new ReentrantLock();
+
+	/**
+	 * Prevent case where {@link VIAConnectProCommunicator#controlProperty(ControllableProperty)} slow down -
+	 * the getMultipleStatistics interval if it's fail to send the cmd
+	 */
+	private static final int controlTelnetTimeout = 3000;
+
+	/**
+	 * Set back to default timeout value in {@link TelnetCommunicator}
+	 */
+	private static final int statisticsTelnetTimeout = 30000;
+
+	/**
+	 * Adapter property: role of a user -
+	 *  Adapter will base on this role to display/hide statistics for a specific role.
+	 */
+	private String configManagement;
+
+	/**
+	 * Retrieves {@code {@link #configManagement }}
+	 *
+	 * @return value of {@link #configManagement}
+	 */
+	public String getConfigManagement() {
+		return configManagement;
+	}
+
+	/**
+	 * Sets {@code configManagement}
+	 *
+	 * @param configManagement the {@code java.lang.String} field
+	 */
+	public void setConfigManagement(String configManagement) {
+		this.configManagement = configManagement;
+	}
+
+	/**
+	 * VIAConnectProCommunicator constructor
+	 */
+	public VIAConnectProCommunicator() {
+		this.setLoginPrompt("Username:");
+		this.setPasswordPrompt("Password:");
+		this.setCommandSuccessList(
+				Collections.singletonList(VIAConnectProConstant.END_COMMAND));
+		this.setCommandErrorList(Collections.singletonList("A"));
+		this.setLoginSuccessList(Collections.singletonList("Login\r\nLogin Successful.\nNow Please send a command:\r\n"));
+		// Instantiate cachedLocalExtendedStatistics
+		cachedLocalExtendedStatistics = new ExtendedStatistics();
+		cachedLocalExtendedStatistics.setStatistics(new HashMap<>());
+		cachedLocalExtendedStatistics.setControllableProperties(new ArrayList<>());
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected void internalInit() throws Exception {
+		super.internalInit();
+		if (logger.isDebugEnabled()) {
+			logger.debug("VIAConnectProCommunicator-internalInit(): Creating telnet session");
+		}
+		// Check if device is reachable, then close the telnet session
+		this.createChannel();
+		if (logger.isDebugEnabled()) {
+			logger.debug("VIAConnectProCommunicator-internalInit(): Closing telnet session");
+		}
+		this.destroyChannel();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected void internalDestroy() {
+		cachedLocalExtendedStatistics.setStatistics(new HashMap<>());
```

Comment:
would it make sense to call getStatistics().clear() and .getControllableProperties().clear() here instead?

Reply:
Yes, I will change it as your comment.